rbx_dom_weak/src/dom.rs

```diff
@@ -277,6 +325,47 @@ impl WeakDom {
 
         instance
     }
+
+    fn ref_into_instancebuilder(
+        &self,
+        ctx: &mut CloneContext,
+        original_ref: Ref,
+    ) -> InstanceBuilder {
+        let instance = self
+            .get_by_ref(original_ref)
+            .expect("Cannot clone an instance that does not exist");
+
+        let builder = InstanceBuilder::new(instance.class.to_string())
+            .with_name(instance.name.to_string())
+            .with_properties(instance.properties.clone());
+
+        let new_ref = builder.referent;
+
+        for uncloned_child in instance.children.iter() {
+            ctx.queue.push_back((new_ref, *uncloned_child))
+        }
+
+        ctx.ref_rewrites.insert(original_ref, new_ref);
+        builder
```

Comment:
This builder *never* has any children and kinda makes me want a fast path in `WeakDom::insert` that doesn't create a `VecDeque` for builders with no children - kind of irrelevant for this PR, but I'll put it on my back burner...